mage/releases/publish.go

```diff
@@ -217,3 +231,35 @@ func listFiles(dir string) []string {
 
 	return names
 }
+
+func generateChecksumFile(path string) (string, error) {
```

Comment:
Let's add some unit tests for this function so that we can verify that it works. Getting build scripts related to publishing can be tricky and it's no fun to cut a release only to realize that there's a bug in the publish logic. Ask me how I know. 😉